### ISSUE TITLE
The tutorial "Running in a Stand-Alone Silo" doesn't appear to work.

### ISSUE THREAD

**Issue by snarlynarwhal** [dotnet/orleans#2614]
I encountered this issue while attempting to do the [Running in a Stand-Alone Silo ](https://dotnet.github.io/orleans/Tutorials/Running-in-a-Stand-alone-Silo.html) tutorial.

The console window for OrleansHost.exe closes immediately upon starting the application. No compiler or run-time errors exist. I completed all of the steps up to and including the "Getting Started" section.

I added the Microsoft.Orleans.Server NuGet package to my grain collection library project and set the project to start OrleansHost.exe on startup. All of the projects have the same NuGet package versions numbers. I created the OrleansConfiguration.xml file and double checked the ProxyingGateway. I read through the tutorial several times now and can't figure out what I am doing wrong. :/

Log file:
`[2017-01-19 02:53:59.376 GMT     1	ERROR  	100105	OrleansSiloHost	]	!!!!!!!!!! ERROR starting Orleans silo name=PJ Exception=
Exc level 0: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config, ILocalDataStore keyStore)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config)
   at Orleans.Runtime.Host.SiloHost.InitializeOrleansSilo()	
Exc level 0: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config, ILocalDataStore keyStore)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config)
   at Orleans.Runtime.Host.SiloHost.InitializeOrleansSilo()
[2017-01-19 02:53:59.385 GMT     1	ERROR  	100105	OrleansSiloHost	]	!!!!!!!!!! ERROR starting Orleans silo name=PJ Exception=
Exc level 0: System.InvalidOperationException: Cannot start silo PJ due to prior initialization error
   at Orleans.Runtime.Host.SiloHost.StartOrleansSilo(Boolean catchExceptions)	
Exc level 0: System.InvalidOperationException: Cannot start silo PJ due to prior initialization error
   at Orleans.Runtime.Host.SiloHost.StartOrleansSilo(Boolean catchExceptions)
`

My grain collection project has those references. I noticed in another thread someone mentioned an OrleansHost.exe.config file.. I don't see that file. o.0

**Comment by attilah** [comment]
@snarlynarwhal since the ```Microsoft.Extensions.DependencyInjection``` package is a transitive dependency to the ```Microsoft.Orleans.Server``` package, I assume that it was not installed by Nuget or Visual Studio, please add this package to the Silo project and try to run the Silo.

Just as a confirmation, I assume you're using the latest published Orleans Nuget packages?

**Comment by attilah** [comment]
I can confirm that following the tutorial step-by-step does not bring the desired result, the project will not run.

I created an issue #2616 to track one of the technical problems I found.

On the other hand the [Minimal Orleans Application](https://dotnet.github.io/orleans/Tutorials/Minimal-Orleans-Application.html) and [Running in a Stand-Alone Silo](https://dotnet.github.io/orleans/Tutorials/Running-in-a-Stand-alone-Silo.html) documentation is outdated so I created a separate issue to track that: #2617

On the fast track to not to block you here are some steps you can do to make the project run:
- Create ```OrleansHost.exe.config``` and set it to "copy to output".

Use this as content:
```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
  <system.diagnostics>
    <trace autoflush="true" indentsize="0">
      <listeners>
        <remove name="*" />
      </listeners>
    </trace>
    <switches>
      <add name="traceLevel" value="2" />
    </switches>
  </system.diagnostics>
  <startup useLegacyV2RuntimeActivationPolicy="true">
    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5"/>
  </startup>
  <runtime>
    <gcServer enabled="true"/>
    <gcConcurrent enabled="false"/>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.WindowsAzure.ServiceRuntime" publicKeyToken="31BF3856AD364E35" culture="neutral"/>
        <bindingRedirect oldVersion="0.0.0.0-2.4.0.0" newVersion="2.4.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.WindowsAzure.Configuration" publicKeyToken="31BF3856AD364E35" culture="neutral"/>
        <bindingRedirect oldVersion="0.0.0.0-2.0.0.0" newVersion="2.0.0.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-7.0.0.0" newVersion="7.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Runtime" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.1.0.0" newVersion="4.1.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Runtime.Extensions" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.1.0.0" newVersion="4.1.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-1.2.0.0" newVersion="1.2.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Runtime.InteropServices" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.1.0.0" newVersion="4.1.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Reflection" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.1.0.0" newVersion="4.1.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Reflection.Metadata" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-1.3.0.0" newVersion="1.3.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

- Uninstall all ```Microsoft.Orleans.*``` packages from ```SiloHost1``` project.
- Install ```Microsoft.Orleans.Client``` package to the ```SiloHost1``` project.

After these steps the tutorial projects are working as expected.

@snarlynarwhal please confirm this so I can close this issue and we're welcoming contributions to #2617 if you feel to do so!

**Comment by snarlynarwhal** [comment]
@attilah I appreciate you investing the issue. I only have three projects: grain collection, grain interfaces, and host which for the former tutorials acted as both the host and the client. For this tutorial, it only has client code. So I only added `Microsoft.Extensions.DependencyInjection` to the grain collection project which starts the silo via `OlreansHost.exe`. And I can confirm that my project does currently use the latest Orleans NuGet packages.

I attempted to follow your instructions, but to no avail - I get the same result as before. To clarify - I put the `OrleansHost.exe.config` file in the collections project since that project starts `OrleansHost.exe`, correct? Also, I did set it to always copy to the output directory. Do I need also set a build action? I also did the uninstall/install of the specified packages.

This time, I received three logs from a single attempt:

**Log 1**

`2017-01-20-02.41.15.296ZZ
ERROR starting Orleans silo name=PJ Exception=
Exc level 0: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config, ILocalDataStore keyStore)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config)
   at Orleans.Runtime.Host.SiloHost.InitializeOrleansSilo()`

**Log 2**

`2017-01-20-02.41.15.301ZZ
ERROR starting Orleans silo name=PJ Exception=
Exc level 0: System.InvalidOperationException: Cannot start silo PJ due to prior initialization error
   at Orleans.Runtime.Host.SiloHost.StartOrleansSilo(Boolean catchExceptions)`

**Log 3**

`[2017-01-20 02:41:15.288 GMT     1	ERROR  	100105	OrleansSiloHost	]	!!!!!!!!!! ERROR starting Orleans silo name=PJ Exception=
Exc level 0: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config, ILocalDataStore keyStore)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config)
   at Orleans.Runtime.Host.SiloHost.InitializeOrleansSilo()	
Exc level 0: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config, ILocalDataStore keyStore)
   at Orleans.Runtime.Silo..ctor(String name, SiloType siloType, ClusterConfiguration config)
   at Orleans.Runtime.Host.SiloHost.InitializeOrleansSilo()
[2017-01-20 02:41:15.298 GMT     1	ERROR  	100105	OrleansSiloHost	]	!!!!!!!!!! ERROR starting Orleans silo name=PJ Exception=
Exc level 0: System.InvalidOperationException: Cannot start silo PJ due to prior initialization error
   at Orleans.Runtime.Host.SiloHost.StartOrleansSilo(Boolean catchExceptions)	
Exc level 0: System.InvalidOperationException: Cannot start silo PJ due to prior initialization error
   at Orleans.Runtime.Host.SiloHost.StartOrleansSilo(Boolean catchExceptions)`

**Comment by attilah** [comment]
Can you share your solution so I can take a look? I did not have this issue, since this package is installed when installing the ```Microsoft.Orleans.Server``` package.

**Comment by snarlynarwhal** [comment]
Of course:
[OrleansTest.zip (Google Drive)](https://drive.google.com/open?id=0B6WVqjhM8m3SQ2Voc0FMVVFndW8)

**Comment by attilah** [comment]
Hi @snarlynarwhal!

The problem you are seeing is because you upgraded all packages to latest (including prerelease) versions, so my pasted ```OrleansHost.exe.config``` will not work.

To make your project work you've to copy the ```assemblyBinding``` element from ```GrainCollection.dll.config``` to ```OrleahsHost.exe.config``` manually.

I'm closing this issue now since it is solved.

**Comment by snarlynarwhal** [comment]
@attilah I updated all of the packages to the latest stable versions. I also copied the `assemblyBinding` element over - no luck (at first).

Then I noticed that it wanted `Microsoft.CodeAnalysis.CSharp` version `1.3.1` not `1.3.2`. So I downgraded to that version and it worked just fine. I just wanted to share this for anyone else that might stumble across this.

Thank you so much for the help! I'd be glad to update the tutorial documentation later this week/weekend.
